xbmc/pvr/timers/PVRTimers.cpp

```diff
@@ -248,6 +248,47 @@ bool CPVRTimers::UpdateEntries(const CPVRTimers &timers)
     UpdateEpgEvent(*timerIt);
   }
 
+  /* go through the timer list and check for parent timers */
+  for (MapTags::const_iterator it = timers.m_tags.begin(); it != timers.m_tags.end(); ++it)
+  {
+    for (VecTimerInfoTag::const_iterator timerIt = it->second->begin(); timerIt != it->second->end(); ++timerIt)
+    {
+      /* check if the parent timer is present */
+      CPVRTimerInfoTagPtr parentTimer = GetByClient((*timerIt)->m_iClientId, (*timerIt)->m_iParentClientIndex);
```

Comment:
Please add
<pre>
unsigned int iRuleId = timersEntry->GetTimerRuleId();
if (iRuleId != PVR_TIMER_NO_PARENT)
</pre>

before doing the expensive call to <code>GetByClientId</code> (look at the implementation to see why it is expensive!)

And, "const as const can" to help the compiler to generate efficient code: <code>const CPVRTimerInfoTagPtr</code>, please.

And even more better <code>const CPVRTimerInfoTagPtr parentTimer(GetByClient(timersEntry->m_iClientId, timersEntry->GetTimerRuleId()));</code>. Always prefer copy ctor (1 call) over standard ctor + assignment operator (2 calls). Even though compilers could optimize this, you never know they actually will do. If you write it like this, compiler will do. :-)